src/tests/simulation/tests/integration_tests/sink/exactly_once_iceberg.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::sync::atomic::Ordering::Relaxed;
+use std::time::Duration;
+
+use anyhow::Result;
+use itertools::Itertools;
+use risingwave_connector::sink::catalog::SinkId;
+use risingwave_simulation::cluster::{Cluster, KillOpts};
+use tokio::time::sleep;
+
+use crate::sink::exactly_once_utils::SimulationTestIcebergExactlyOnceSink;
+use crate::sink::utils::{
+    CREATE_SINK, CREATE_SOURCE, DROP_SINK, DROP_SOURCE, SimulationTestSource,
+    start_sink_test_cluster,
+};
+use crate::{assert_eq_with_err_returned as assert_eq, assert_with_err_returned as assert};
+
+#[tokio::test]
+async fn test_exactly_once_sink_basic() -> Result<()> {
+    let err_rate_list = vec![0.1, 0.1, 0.1, 0.1, 0.1];
+    test_exactly_once_sink_inner(err_rate_list, false).await
+}
+
+#[tokio::test]
+#[should_panic]
+async fn test_exactly_once_sink_should_panic() {
+    // If, during a re-commit, a new snapshot_id is generated instead of using the previously persisted snapshot_id, it can lead to duplicate data when consecutive "23" appears in err_events.
+    let err_rate_list = vec![0.0, 0.0, 0.3, 0.3, 0.0];
+    let _ = test_exactly_once_sink_inner(err_rate_list, true).await;
+}
```

Comment:
If the error rate is not 1, how can it ensure the test must panic?

Reply:
After some offline discussion, removed the should panic test.